producers/http/handlers.go

```diff
@@ -51,7 +51,13 @@ func nodeHandler(p *producerImpl) http.HandlerFunc {
 func containersHandler(p *producerImpl) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		cm := []string{}
-		containerMetrics, err := p.store.GetByRegex(regexp.QuoteMeta(producers.ContainerMetricPrefix) + ".*")
+		containerMetrics, err := p.store.GetByRegex(
+			fmt.Sprintf(
+				"(%s|%s).*",
+				regexp.QuoteMeta(producers.ContainerMetricPrefix),
+				regexp.QuoteMeta(producers.AppMetricPrefix),
+			),
+		)
```

Comment:
It's not clear to me exactly how this works. I don't see how we get container metrics for nested containers using this.

Reply:
What I understand about this is that because all of the app metrics should have the container ID set in the dimensions, even those container IDs that the mesos agent fails to provide from /containers. This allows the container IDs to be successfully collected.

Reply:
We don't get container metrics for nested containers (I don't know of any way to do so at present) - hitting `/metrics/v0/containers/<nested-container-id>` will result in a 204 empty response. What this change does is make sure that both the nested container and its parent have their IDs in `/metrics/v0/containers`.

Reply:
So we now have nested container IDs in `/metrics/v0/containers/`, but hitting `/metrics/v0/containers/<nested-container-id>` will return a 204? What did it do before this change? Return a 404 because we didn't even have a reference to the container? This seems strange to me, but if mesos doesn't return any metrics about nested containers, I guess there's not much we can do about it -- and returning 202 No Content, seems like the proper response.

So I guess the real point of this change is so we can now hit `/metrics/v0/containers/<nested-container-id>/app` and get some app metrics for this container. Before this change, this wasn't possible because we weren't tracking the nested container IDs under `/metrics/v0/containers/` at all.

Is this an accurate assessment?

Reply:
That's almost accurate - it's always been possible to hit `/metrics/v0/containers/<nested-container-id>/app` and get metrics. But the user would have had to go figure out what the nested container's ID was themselves instead of finding it in the `/metrics/v0/containers` endpoint.

Before this PR, hitting `/metrics/v0/containers/<nested-container-id>` endpoint would have returned a 404. This PR literally only adds nested container IDs to the container endpoint using data we already had available.